infra.bs

```diff
@@ -941,6 +941,69 @@ ordering will not match any particular alphabet or lexicographic order, particul
 
 <hr>
 
+<p>The <dfn export>code unit substring</dfn> from index <var>start</var> with length
+<var>length</var> within a <a>string</a> <var>string</var> is determined as follows:
+
+<ol>
+ <li><p><a>Assert</a>: <var>start</var> and <var>length</var> are greater than 0.</p></li>
+
+ <li><p><a>Assert</a>: <var>start</var> + <var>length</var> is less than or equal to
+ <var>string</var>'s <a for=string>length</a>.</p></li>
+
+ <li><p>Let <var>result</var> be the empty string.</p></li>
+
+ <li><p><a for="set">For each</a> <var>i</var> in <a lt="the exclusive range">the range</a> from
+ <var>start</var> to <var>start</var> + <var>length</var>, exclusive: append the <var>i</var>th
+ <a>code unit</a> of <var>string</var> to <var>result</var>.</p></li>
+
+ <li><p>Return <var>result</var>.</li>
+</ol>
+
+<p>The <dfn export lt="code unit substring by indices">code unit substring</dfn> from indices
+<var>start</var> to <var>end</var> inclusive within a <a>string</a> <var>string</var> is the <a>code
+unit substring</a> from <var>start</var> with length <var>end</var> &minus; <var>start</var> within
+<var>string</var>.
+
+<p class="example" id="example-code-unit-substring">The <a>code unit substring</a> from index 1 with
+length 3 within "<code>Hello world</code>" is "<code>ell</code>". This can also be expressed as the
+<a lt="code unit substring by indices">code unit substring</a> from indices 1 to 4 inclusive.
```

Comment:
```suggestion
<a lt="code unit substring by indices">code unit substring</a> from indices 1 to 4, inclusive.
```

Reply:
I didn't do this because then you need a lot more commas. In particular you can no longer say "Concatenate the substring from indices X to Y inclusive within Z with W". Instead you need "Concatenate the substring from indices Y to Z, inclusive, within Z, with W" which feels a good bit more awkward.

Reply:
I see, I'm not sure I would have a good intuition as to when to include and when to omit them, as thus far I think I have consistently used a comma before inclusive (and sometimes after, if there's not a period). It does seem reasonable to omit it in a longer sentence like that though.